### PR TITLE
Introduce isDeprecated to CallableParametersAcceptor

### PR DESCRIPTION
**Pull request by kamil-zacek**

### CODE REVIEW

File: src/Type/ClosureType.php [hunk 1]
```diff
@@ -143,6 +143,11 @@ public function isPure(): TrinaryLogic
 		return $certainCount > 0 ? TrinaryLogic::createNo() : TrinaryLogic::createMaybe();
 	}
 
+	public function isDeprecated(): TrinaryLogic
+	{
+		return TrinaryLogic::createNo();
```
Comment (ondrejmirtes) [inline src/Type/ClosureType.php]: This is not the case. First-class callables like `Foo::method(...)` are resolved to ClosureType so the deprecated-ness should be carried over.

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]
Also please review all `new CallableType` to make sure we haven't missed a place where it should be added.

File: src/Type/ClosureType.php (hunk 1 above)
Comment (kamil-zacek) [reply src/Type/ClosureType.php]: I am not sure how to deal with it. Do you think the class ClosureType itself should have a member "isDeprecated" which will be in the constructor and resolved during the creation of instance of ClosureType itself?

**Comment by staabm** [comment]
I guess rhis was rebased against 2.1.x but the PR target needs still to be changed to 2.1.x

**Comment by kamil-zacek** [comment]
> I guess rhis was rebased against 2.1.x but the PR target needs still to be changed to 2.1.x

Yes, I know it. I will close and reopen PR when I receive reply to my previous question.

**Pull request closed without merging**